cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -192,7 +192,7 @@ func start(log log.Component, config config.Component, telemetry telemetry.Compo
 	demux := aggregator.InitAndStartAgentDemultiplexer(log, forwarder, opts, hname)
 	demux.AddAgentStartupTelemetry(fmt.Sprintf("%s - Datadog Cluster Agent", version.AgentVersion))
 
-	le, err := leaderelection.GetLeaderEngine()
+	le, err := leaderelection.GetLeaderEngine(mainCtx)
```

Comment:
This present the typical singleton-with-parameters issue. As the `leaderelection` is initialized once at startup, it'd be perhaps better to have a `InitaliziteLeaderElection` method and then have a no parameter `Get`.

Reply:
Since leader election can also be initialized inside a check (because normal Agents can also run a leader election), it can be more complex. Inside these checks, we don't have access to the main context.

We could also initialize leader election in the agent start command but I think perhaps it could be overkill and error prone.

So I propose a simple function `CreateLeaderEngine` that sets the context of the leader election engine.

Reply:
The checks that are checking for leader election are checking it when running in the Cluster Agent. If not they're cluster checks and are guaranteed to run only once.

There's an option for node Agents to have leader election as well. Outside of not being recommended, it does not really change the structure of the interactions.

The goal is to have the LeaderEngineCreation happen once in the `cmd/*` paths (for Cluster Agent and Agent), and users are just re-using this instance.
The instance creation should not happen anywhere else.